### PR TITLE
Add support for accessing `x86_64` registers

### PR DESCRIPTION
**Pull request by EliaGeretto**
This PR adds support for accessing `x86_64` registers. It requires `libunwind >= 1.6.0`, which contains the fix for the issue described in #16. It also contains a small fix to the build system which allows building with `-rc` releases of `libunwind`.

### CODE REVIEW

File: unwind/src/lib.rs [hunk 1]
```diff
@@ -238,6 +238,9 @@ impl RegNum {
     pub const SP: RegNum = RegNum(UNW_REG_SP);
 }
 
+#[cfg(any(target_arch = "x86_64", doc))]
```
Comment (sfackler) [inline unwind/src/lib.rs:241]: Will generating docs on another architecture actually work given that the unwind_sys constants these refer to won't exist?

Comment (EliaGeretto) [reply unwind/src/lib.rs:241]: Yes, you are right. Removing `doc` then.

File: unwind/src/lib.rs [hunk 2]
```diff
@@ -238,6 +238,9 @@ impl RegNum {
     pub const SP: RegNum = RegNum(UNW_REG_SP);
 }
 
+#[cfg(any(target_arch = "x86_64"))]
```
Comment (sfackler) [inline unwind/src/lib.rs:241]:
```suggestion
#[cfg(target_arch = "x86_64")]
```

**Comment by sfackler** [comment]
Looks like the build script change may have broken some of the version detection logic?

**Comment by EliaGeretto** [comment]
If you are talking about the failed CI checks, they are expected to fail with version 1.4 on `x86_64`. The required version, the one with the fix for #16, is 1.6.

**Comment by sfackler** [comment]
I certainly do not expect to see segfaults when using 1.4. That does not appear to be the case on master: https://github.com/sfackler/rstack/runs/3077148196?check_suite_focus=true

**Comment by EliaGeretto** [comment]
The segfault is a symptom of the bug described in #16. It happens with 1.4 in the test case I added in this PR. The cause is a self-reference in the unwind cursor in that version of the library. That reference leads to a use-after-free, which was fixed in 1.6.

**Comment by sfackler** [comment]
It sounds like we would need to disable this test on those versions then.

**Comment by EliaGeretto** [comment]
Closing this in favor of #22.

**Pull request closed without merging**